File: count-primes/count-primes.py
class Solution:
    def countPrimes(self, n: int) -> int:
        '''
        Counts the number of prime numbers less than a non-negative int n
        Returns the count of primes
        '''
        primes = [True for i in range(n+1)] #Time Complexity O(n), Space Complexity O(n)
        p = 2
        count = 0
        while p * p < n: #Only need to check up to sqrt(n)
            if primes[p] == True:
                for i in range(p * p, n + 1, p): #Mark larger products of seen value
                    primes[i] = False
            p += 1
        for p in range(2, n): #Count remaining prime spaces
            if primes[p]:
                count += 1
        return count

File: count-primes/count-primes.java
class Solution 
{
    public int countPrimes(int n) 
    {
        // Create array to hold list of primes
        boolean[] notPrime = new boolean[n];
        int count = 0;
        
        // Run through every number i -> n and check if seen as prime
        for (int i = 2; i < n; i++)
        {
            if (!notPrime[i]) // Found a prime, remove divisors of it
            {
                count += 1;
                for (int j = 2; i*j < n; j++)
                    notPrime[i*j] = true;
            }
        }
        return count;
    }
}